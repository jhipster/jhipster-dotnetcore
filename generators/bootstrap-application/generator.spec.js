import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'bootstrap-application';

describe('SubGenerator bootstrap-application of dotnetcore JHipster blueprint', () => {
  describe('run', () => {
    beforeAll(async function () {
      await helpers
        .runJHipster(SUB_GENERATOR)
        .withJHipsterConfig()
        .withOptions({
          ignoreNeedlesError: true,
          blueprints: 'dotnetcore',
        })
        .withJHipsterLookup()
        .withParentBlueprintLookup();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
    it('application should match snapshot', () => {
      expect({ ...result.generator.sharedData.getApplication(), user: undefined }).toMatchSnapshot();
    });
  });
});
