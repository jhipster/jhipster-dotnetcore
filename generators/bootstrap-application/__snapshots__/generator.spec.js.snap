// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator bootstrap-application of dotnetcore JHipster blueprint > run > application should match snapshot 1`] = `
{
  "SERVER_SRC_DIR": "src/",
  "SERVER_TEST_DIR": "test/",
  "applicationType": "monolith",
  "applicationTypeAny": true,
  "applicationTypeGateway": false,
  "applicationTypeMicroservice": false,
  "applicationTypeMonolith": true,
  "authenticationType": "jwt",
  "authenticationTypeJwt": true,
  "authenticationTypeOauth2": false,
  "authenticationTypeSession": false,
  "authenticationUsesCsrf": false,
  "backendName": ".Net",
  "backendType": ".Net",
  "backendTypeJavaAny": false,
  "backendTypeSpringBoot": false,
  "baseName": "jhipster",
  "blueprints": [
    {
      "name": "generator-jhipster-dotnetcore",
      "version": "3.9.1",
    },
  ],
  "buildTool": "maven",
  "buildToolGradle": false,
  "buildToolMaven": true,
  "buildToolUnknown": false,
  "cacheProvider": "ehcache",
  "cacheProviderAny": true,
  "cacheProviderCaffeine": false,
  "cacheProviderEhcache": true,
  "cacheProviderHazelcast": false,
  "cacheProviderInfinispan": false,
  "cacheProviderMemcached": false,
  "cacheProviderNo": false,
  "cacheProviderRedis": false,
  "camelizedBaseName": "jhipster",
  "capitalizedBaseName": "Jhipster",
  "clientDistDir": "src/Jhipster/ClientApp/dist/",
  "clientFramework": "angular",
  "clientFrameworkAngular": true,
  "clientFrameworkAny": true,
  "clientFrameworkNo": false,
  "clientFrameworkReact": false,
  "clientFrameworkVue": false,
  "clientPackageManager": "npm",
  "clientRootDir": "src/Jhipster/ClientApp/",
  "clientSrcDir": "src/Jhipster/ClientApp/src/",
  "clientTestDir": "src/Jhipster/ClientApp/test/",
  "clientTestFrameworks": undefined,
  "clientTestFrameworksAny": true,
  "clientTestFrameworksCypress": false,
  "clientTestProject": "Jhipster/ClientApp/test/",
  "clientTheme": "none",
  "clientThemeDark": false,
  "clientThemeLight": false,
  "clientThemeNone": true,
  "clientThemePrimary": false,
  "clientThemeVariant": "",
  "communicationSpringWebsocket": false,
  "cqrsEnabled": undefined,
  "cucumberTests": false,
  "cypressTests": false,
  "dasherizedBaseName": "jhipster",
  "databaseMigration": undefined,
  "databaseMigrationLiquibase": false,
  "databaseType": "sqllite",
  "databaseTypeAny": true,
  "databaseTypeCassandra": false,
  "databaseTypeCouchbase": false,
  "databaseTypeMongodb": false,
  "databaseTypeNeo4j": false,
  "databaseTypeNo": false,
  "databaseTypeSql": false,
  "databaseTypeSqllite": true,
  "devDatabaseType": "postgresql",
  "devDatabaseTypeH2Any": undefined,
  "devServerPort": undefined,
  "dockerContainers": {
    "apachepulsar/pulsar": "apachepulsar/pulsar:3.0.1",
    "cassandra": "cassandra:3.11.14",
    "cassandraImage": "cassandra",
    "cassandraTag": "3.11.14",
    "confluentinc/cp-kafka": "confluentinc/cp-kafka:7.5.1",
    "confluentinc/cp-zookeeper": "confluentinc/cp-zookeeper:7.5.1",
    "consul": "docker.io/bitnami/consul:1.16.2",
    "consulConfigLoader": "jhipster/consul-config-loader:v0.4.1",
    "consulConfigLoaderImage": "jhipster/consul-config-loader",
    "consulConfigLoaderTag": "v0.4.1",
    "consulImage": "docker.io/bitnami/consul",
    "consulTag": "1.16.2",
    "couchbase": "couchbase/server:7.2.2",
    "couchbase/server": "couchbase/server:7.2.2",
    "couchbaseImage": "couchbase/server",
    "couchbaseTag": "7.2.2",
    "docker.io/bitnami/consul": "docker.io/bitnami/consul:1.16.2",
    "eclipse-temurin": "eclipse-temurin:17-jre-focal",
    "eclipseTemurin": "eclipse-temurin:17-jre-focal",
    "eclipseTemurinImage": "eclipse-temurin",
    "eclipseTemurinTag": "17-jre-focal",
    "elasticsearch": "docker.elastic.co/elasticsearch/elasticsearch:8.7.1",
    "elasticsearchImage": "docker.elastic.co/elasticsearch/elasticsearch",
    "elasticsearchTag": "8.7.1",
    "grafana": "grafana/grafana:10.2.0",
    "grafana/grafana": "grafana/grafana:10.2.0",
    "grafanaImage": "grafana/grafana",
    "grafanaTag": "10.2.0",
    "grafanaWatcher": "quay.io/coreos/grafana-watcher:v0.0.8",
    "grafanaWatcherImage": "quay.io/coreos/grafana-watcher",
    "grafanaWatcherTag": "v0.0.8",
    "hazelcast": "hazelcast/management-center:5.3.3",
    "hazelcast/management-center": "hazelcast/management-center:5.3.3",
    "hazelcastImage": "hazelcast/management-center",
    "hazelcastTag": "5.3.3",
    "javaJre": "eclipse-temurin:17-jre-focal",
    "javaJreImage": "eclipse-temurin",
    "javaJreTag": "17-jre-focal",
    "jenkins": "jenkins/jenkins:lts-jdk11",
    "jenkins/jenkins": "jenkins/jenkins:lts-jdk11",
    "jenkinsImage": "jenkins/jenkins",
    "jenkinsTag": "lts-jdk11",
    "jhipster/consul-config-loader": "jhipster/consul-config-loader:v0.4.1",
    "jhipster/jhipster-control-center": "jhipster/jhipster-control-center:v0.5.0",
    "jhipster/jhipster-registry": "jhipster/jhipster-registry:v7.4.0",
    "jhipsterControlCenter": "jhipster/jhipster-control-center:v0.5.0",
    "jhipsterControlCenterImage": "jhipster/jhipster-control-center",
    "jhipsterControlCenterTag": "v0.5.0",
    "jhipsterRegistry": "jhipster/jhipster-registry:v7.4.0",
    "jhipsterRegistryImage": "jhipster/jhipster-registry",
    "jhipsterRegistryTag": "v7.4.0",
    "kafka": "confluentinc/cp-kafka:7.5.1",
    "kafkaImage": "confluentinc/cp-kafka",
    "kafkaTag": "7.5.1",
    "keycloak": "quay.io/keycloak/keycloak:22.0.5",
    "keycloakImage": "quay.io/keycloak/keycloak",
    "keycloakTag": "22.0.5",
    "mariadb": "mariadb:11.1.2",
    "mariadbImage": "mariadb",
    "mariadbTag": "11.1.2",
    "mcr.microsoft.com/mssql/server": "mcr.microsoft.com/mssql/server:2019-CU16-GDR1-ubuntu-20.04",
    "memcached": "memcached:1.6.22-alpine",
    "memcachedImage": "memcached",
    "memcachedTag": "1.6.22-alpine",
    "mongo": "mongo:7.0.2",
    "mongoImage": "mongo",
    "mongoTag": "7.0.2",
    "mongodb": "mongo:7.0.2",
    "mongodbImage": "mongo",
    "mongodbTag": "7.0.2",
    "mssql": "mcr.microsoft.com/mssql/server:2019-CU16-GDR1-ubuntu-20.04",
    "mssqlImage": "mcr.microsoft.com/mssql/server",
    "mssqlTag": "2019-CU16-GDR1-ubuntu-20.04",
    "mysql": "mysql:8.2.0",
    "mysqlImage": "mysql",
    "mysqlTag": "8.2.0",
    "neo4J": "neo4j:5.13.0",
    "neo4JImage": "neo4j",
    "neo4JTag": "5.13.0",
    "neo4j": "neo4j:5.13.0",
    "openzipkin/zipkin": "openzipkin/zipkin:2.24",
    "postgres": "postgres:16.0",
    "postgresImage": "postgres",
    "postgresTag": "16.0",
    "postgresql": "postgres:16.0",
    "postgresqlImage": "postgres",
    "postgresqlTag": "16.0",
    "prom/alertmanager": "prom/alertmanager:v0.26.0",
    "prom/prometheus": "prom/prometheus:v2.47.2",
    "prometheus": "prom/prometheus:v2.47.2",
    "prometheusAlertmanager": "prom/alertmanager:v0.26.0",
    "prometheusAlertmanagerImage": "prom/alertmanager",
    "prometheusAlertmanagerTag": "v0.26.0",
    "prometheusImage": "prom/prometheus",
    "prometheusOperator": "quay.io/coreos/prometheus-operator:v0.42.1",
    "prometheusOperatorImage": "quay.io/coreos/prometheus-operator",
    "prometheusOperatorTag": "v0.42.1",
    "prometheusTag": "v2.47.2",
    "pulsar": "apachepulsar/pulsar:3.0.1",
    "pulsarImage": "apachepulsar/pulsar",
    "pulsarTag": "3.0.1",
    "quay.io/coreos/grafana-watcher": "quay.io/coreos/grafana-watcher:v0.0.8",
    "quay.io/coreos/prometheus-operator": "quay.io/coreos/prometheus-operator:v0.42.1",
    "quay.io/keycloak/keycloak": "quay.io/keycloak/keycloak:22.0.5",
    "redis": "redis:7.2.2",
    "redisImage": "redis",
    "redisTag": "7.2.2",
    "sonar": "sonarqube:10.2.1-community",
    "sonarImage": "sonarqube",
    "sonarTag": "10.2.1-community",
    "sonarqube": "sonarqube:10.2.1-community",
    "sonarqubeImage": "sonarqube",
    "sonarqubeTag": "10.2.1-community",
    "swaggerEditor": "swaggerapi/swagger-editor:latest",
    "swaggerEditorImage": "swaggerapi/swagger-editor",
    "swaggerEditorTag": "latest",
    "swaggerapi/swagger-editor": "swaggerapi/swagger-editor:latest",
    "zipkin": "openzipkin/zipkin:2.24",
    "zipkinImage": "openzipkin/zipkin",
    "zipkinTag": "2.24",
    "zookeeper": "confluentinc/cp-zookeeper:7.5.1",
    "zookeeperImage": "confluentinc/cp-zookeeper",
    "zookeeperTag": "7.5.1",
  },
  "dockerServicesDir": "docker/",
  "documentationArchiveUrl": "https://www.jhipster.tech/documentation-archive/v8.0.0",
  "dtoSuffix": "Dto",
  "embeddableLaunchScript": undefined,
  "enableGradleEnterprise": false,
  "enableHibernateCache": true,
  "enableI18nRTL": false,
  "enableSwaggerCodegen": false,
  "enableTranslation": true,
  "endpointPrefix": "",
  "entitySuffix": "",
  "fakerSeed": undefined,
  "frontendAppName": "jhipsterApp",
  "gatewayServerPort": undefined,
  "gatlingTests": false,
  "generateAuthenticationApi": true,
  "generateBuiltInAuthorityEntity": true,
  "generateBuiltInUserEntity": true,
  "generateInMemoryUserCredentials": false,
  "generateUserManagement": true,
  "gradleEnterpriseHost": undefined,
  "gradleVersion": "8.4",
  "hipster": "jhipster_family_member_3",
  "humanizedBaseName": "Jhipster",
  "imperativeOrReactive": "imperative",
  "incrementalChangelog": undefined,
  "javaDependencies": {
    "archunit-junit5": "1.1.0",
    "awaitility": "4.2.0",
    "blockhound-junit-platform": "1.0.8.RELEASE",
    "cassandra": "4.15.0",
    "checkstyle": "10.12.4",
    "checksum-maven-plugin": 1.11,
    "common-custom-user-data-gradle-plugin": "1.12",
    "commons-beanutils": "1.9.4",
    "frontend-maven-plugin": "1.14.2",
    "gatling": "3.9.5",
    "gatling-gradle": "3.9.5.6",
    "gatling-maven-plugin": "4.6.0",
    "git-commit-id-maven-plugin": "7.0.0",
    "gradle-enterprise": "3.15.1",
    "gradle-git-properties": "2.4.1",
    "gradle-liquibase": "2.2.0",
    "gradle-modernizer-plugin": "1.9.0",
    "gradle-sonarqube": "4.4.1.3373",
    "h2": "2.2.224",
    "hazelcast-hibernate53": "5.1.0",
    "hazelcast-spring": "5.3.5",
    "hibernate": "6.2.13.Final",
    "jackson-databind-nullable": "0.2.6",
    "jacoco-maven-plugin": "0.8.11",
    "jaxb-runtime": "4.0.4",
    "jib-maven-plugin": "3.4.0",
    "junit-platform-launcher": "1.10.0",
    "lifecycle-mapping": "1.0.0",
    "liquibase": "4.24.0",
    "mapstruct": "1.5.5.Final",
    "maven-antrun-plugin": "3.1.0",
    "maven-checkstyle-plugin": "3.3.1",
    "maven-clean-plugin": "3.3.2",
    "maven-compiler-plugin": "3.11.0",
    "maven-eclipse-plugin": 2.1,
    "maven-enforcer-plugin": "3.4.1",
    "maven-failsafe-plugin": "3.2.1",
    "maven-idea-plugin": "2.2.1",
    "maven-jar-plugin": "3.3.0",
    "maven-javadoc-plugin": "3.6.0",
    "maven-resources-plugin": "3.3.1",
    "maven-site-plugin": "3.12.1",
    "maven-surefire-plugin": "3.2.1",
    "maven-war-plugin": "3.4.0",
    "micrometer-context-propagation": "1.0.6",
    "modernizer-maven-plugin": "2.7.0",
    "node-gradle": "7.0.1",
    "nohttp-checkstyle": "0.0.11",
    "openapi-generator-maven-plugin": "7.0.1",
    "picocli": "4.7.5",
    "properties-maven-plugin": "1.2.1",
    "sonar-maven-plugin": "3.10.0.2594",
    "spotless-gradle-plugin": "6.22.0",
    "spotless-maven-plugin": "2.40.0",
    "spring-boot": "3.1.5",
    "spring-pulsar": "0.2.0",
    "typesafe": "1.4.3",
    "validation-api": "3.0.2",
  },
  "javaPackageSrcDir": "src/main/java/com/mycompany/myapp/",
  "javaPackageTestDir": "src/test/java/com/mycompany/myapp/",
  "javaVersion": "17",
  "jhiPrefix": "jhi",
  "jhiPrefixCapitalized": "Jhi",
  "jhiPrefixDashed": "jhi",
  "jhiTablePrefix": "jhi",
  "jhipsterDotnetVersion": undefined,
  "jhipsterVersion": "8.0.0",
  "jwtSecretKey": undefined,
  "kebabCasedBaseName": "jhipster",
  "languages": [
    "en",
    "fr",
  ],
  "languagesDefinition": [
    {
      "angularLocale": "en",
      "dayjsLocale": "en",
      "displayName": "English",
      "fakerjsLocale": "en",
      "javaLocaleMessageSourceSuffix": "en",
      "languageTag": "en",
      "name": "English",
      "rtl": false,
    },
    {
      "angularLocale": "fr",
      "dayjsLocale": "fr",
      "displayName": "Français",
      "fakerjsLocale": "fr",
      "javaLocaleMessageSourceSuffix": "fr",
      "languageTag": "fr",
      "name": "French",
      "rtl": false,
    },
  ],
  "lowercaseBaseName": "jhipster",
  "mainClass": "JhipsterApp",
  "mainClientAppDir": "Jhipster/ClientApp/src/",
  "mainClientDir": "Jhipster/ClientApp/",
  "mainJavaDir": "src/main/java/",
  "mainJavaPackageDir": "src/main/java/com/mycompany/myapp/",
  "mainJavaResourceDir": "src/main/resources/",
  "mainProjectDir": "Jhipster/",
  "messageBroker": "no",
  "messageBrokerAny": false,
  "messageBrokerKafka": false,
  "messageBrokerPulsar": false,
  "microfrontend": false,
  "microfrontends": undefined,
  "modelSuffix": "Model",
  "namespace": "Jhipster",
  "nativeLanguage": "en",
  "nativeLanguageDefinition": {
    "angularLocale": "en",
    "dayjsLocale": "en",
    "displayName": "English",
    "fakerjsLocale": "en",
    "javaLocaleMessageSourceSuffix": "en",
    "languageTag": "en",
    "name": "English",
    "rtl": false,
  },
  "nodeDependencies": {
    "concurrently": "8.2.2",
    "husky": "8.0.3",
    "lint-staged": "15.0.2",
    "npm": "10.2.2",
    "prettier": "3.0.3",
    "prettier-plugin-java": "2.3.1",
    "prettier-plugin-packagejson": "2.4.6",
    "wait-on": "7.0.1",
  },
  "nodePackageManager": "npm",
  "nodeVersion": "18.18.2",
  "packageFolder": "com/mycompany/myapp/",
  "packageInfoJavadocs": [],
  "packageName": "com.mycompany.myapp",
  "pages": [],
  "pascalizedBaseName": "Jhipster",
  "prettierExtensions": "md,json,yml,html,cjs,mjs,js,ts,tsx,css,scss,java",
  "prettierJava": undefined,
  "primaryKeyType": "long",
  "prodDatabaseType": "postgresql",
  "prodDatabaseTypeMariadb": undefined,
  "prodDatabaseTypeMssql": undefined,
  "prodDatabaseTypePostgresql": undefined,
  "projectDescription": "Description for JHipster",
  "reactive": false,
  "relativeMainAppDir": "ClientApp/src/",
  "relativeMainClientDir": "ClientApp/",
  "relativeMainTestDir": "ClientApp/test/",
  "rememberMeKey": undefined,
  "searchEngine": "no",
  "searchEngineAny": false,
  "searchEngineCouchbase": false,
  "searchEngineElasticsearch": false,
  "searchEngineNo": true,
  "serverPort": 8080,
  "serverPortSecured": 8081,
  "serviceDiscoveryAny": false,
  "serviceDiscoveryConsul": false,
  "serviceDiscoveryEureka": false,
  "serviceDiscoveryType": "no",
  "skipCheckLengthOfIdentifier": false,
  "skipClient": undefined,
  "skipCommitHook": true,
  "skipFakeData": false,
  "skipJhipsterDependencies": undefined,
  "skipServer": undefined,
  "skipUserManagement": false,
  "solutionName": "Jhipster",
  "srcMainDir": "src/main/",
  "srcMainJava": "src/main/java/",
  "srcMainResources": "src/main/resources/",
  "srcMainWebapp": "src/main/webapp/",
  "srcTestDir": "src/test/",
  "srcTestJava": "src/test/java/",
  "srcTestJavascript": "src/test/javascript/",
  "srcTestResources": "src/test/resources/",
  "temporaryDir": "tmp/",
  "testFrameworks": [],
  "testJavaDir": "src/test/java/",
  "testJavaPackageDir": "src/test/java/com/mycompany/myapp/",
  "testProjectDir": "Jhipster.Test/",
  "testResourceDir": "src/test/resources/",
  "upperFirstCamelCaseBaseName": "Jhipster",
  "useNpmWrapper": false,
  "user": undefined,
  "websocket": "no",
  "withAdminUi": false,
  "withTerraformAzureScripts": undefined,
}
`;

exports[`SubGenerator bootstrap-application of dotnetcore JHipster blueprint > run > should succeed 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
}
`;
